src/Integration/Aws/SimpleS3/src/SimpleS3Client.php

```diff
@@ -21,8 +26,146 @@ public function getUrl(string $bucket, string $key): string
         return $this->getEndpoint($uri, [], null);
     }
 
-    /*
-     * TODO Add get() and put() which supports large files
-     * TODO add has()
+    public function download(string $bucket, string $key): ResultStream
+    {
+        return $this->getObject(['Bucket' => $bucket, 'Key' => $key])->getBody();
+    }
+
+    public function has(string $bucket, string $key): bool
+    {
+        return $this->objectExists(['Bucket' => $bucket, 'Key' => $key])->isSuccess();
+    }
+
+    /**
+     * @param string|resource|callable|iterable $object
+     * @param array{
+     *   ACL?: \AsyncAws\S3\Enum\ObjectCannedACL::*,
+     *   CacheControl?: string,
+     *   ContentLength?: string,
+     *   ContentType?: string,
+     *   Metadata?: array<string, string>,
+     *   PartSize?: int,
+     * } $options
      */
+    public function upload(string $bucket, string $key, $object, array $options = []): void
+    {
+        $megabyte = 1024 * 1024;
+        // Split the stream in 1MB chunk
+        $stream = $this->getStream($object, 1 * $megabyte);
+
+        if (!empty($options['ContentLength'])) {
+            $contentLength = (int) $options['ContentLength'];
+        } else {
+            $contentLength = $stream->length();
```

Comment:
could be
```php
$contentLength = (int)($options['ContentLength'] ?? $stream->length());
```

Reply:
No, after your line, then $contentLength will be an int. 

`$stream->length()` returns `?int`